### PR TITLE
Can create user and session 

### PR DESCRIPTION
**Pull request by tyladevon**
### Sidebar Checklist

-  [x] Request reviewers
-  [x] Assign yourself and other contributors
-  [x] Link to project

### Issues Resolved
Resolves # 31 and 32

### Problem Addressed
Needed the ability to create a session along with creating a user 
### Solution Implemented
This overrides the CreateUser Mutation through GraphQL, but enables FE to create user and session through RESTful BE api endpoint.
### Other Notes
One spec has been created for testing basic function of create user and session. Will need to make spec more robust. #31 and #32 will not be considered complete until more robust testing is done

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -36,9 +36,14 @@ class Application < Rails::Application
     # Only loads a smaller set of middleware suitable for API only apps.
     # Middleware like session, flash, cookies can be added back manually.
     # Skip views, helpers and assets when generating a new resource.
-    config.api_only = true
+    config.api_only = false
 
     # For GraphiQL, in conjunction with above requirement of sprockets/railtie
     config.middleware.use Rack::MethodOverride
+    config.middleware.use ActionDispatch::Flash
```
Comment (rer7891) [inline config/application.rb:43]: Nice job configuring middleware to add sessions back into the app

File: app/controllers/graphql_controller.rb [hunk 2]
```diff
@@ -21,6 +20,10 @@ def execute
 
   private
 
+  def current_user
+    context[:current_user]
```
Comment (rer7891) [inline app/controllers/graphql_controller.rb:24]: Thanks for adding a helper method and figuring out the context hash.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,8 @@
 class ApplicationController < ActionController::API
+	include ActionController::Helpers	
+	helper_method :current_user
+	
+	def current_user
+		@current_user ||= User.find(session[:current_user]) if session[:current_user]
```
Comment (rer7891) [inline app/controllers/application_controller.rb:6]: Thanks for getting this helper method as well!

File: spec/requests/api/v1/users_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+		post '/api/v1/users', params: params
+
+		post '/graphql', params: { query: query }
+
+		expect(response).to be_successful
+		JSON.parse(response.body, symbolize_names: true)
+	end
+
+	def query
```
Comment (rer7891) [inline spec/requests/api/v1/users_spec.rb:20]: This is not necessary but we can also take it out later.

File: app/controllers/api/v1/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+		user = User.create(user_params)
+		if user.save
+			session[:current_user] = user.id
+		end
+	end
+
+	private 
+
+	def user_params 
+		params.permit(:first_name, :last_name, :email, :google_token)
```
Comment (wrenisit) [inline app/controllers/api/v1/users_controller.rb:12]: Thank you for using strong params. This is great.

File: config/routes.rb [hunk 6]
```diff
@@ -5,4 +5,10 @@
 
   post "/graphql", to: "graphql#execute"
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  
+  namespace :api do
```
Comment (wrenisit) [inline config/routes.rb:9]: Thank you for name spacing the API. Very nice!

**Pull request merged**